fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/MatchCandidate.java

```diff
@@ -273,6 +273,7 @@ static Iterable<MatchCandidate> fromIndexDefinition(@Nonnull final RecordMetaDat
 
         switch (indexType) {
             case IndexTypes.VALUE:
+            case IndexTypes.VERSION:
```

Comment:
This is more of a design question, why do we need a dedicated index type for version? In other words, isn't serialising the record version along with the record an internal function that can be applied to _any_ value index?